dataactvalidator/validation_handlers/validator.py

```diff
@@ -453,6 +453,83 @@ def rule_exists_in_table(cls, data, value, rule, datatype, interfaces, record):
+        """ Check that set of values exists in specified table, rule_text_1 is table, rule_text_2 is dict mapping
+        columns in record to columns in domain values table """
+        # Load mapping from record fields to domain value fields
+        fieldMap = json.loads(rule.rule_text_2)
+        # Get values for fields in record into new dict between table columns and values to check for
+        valueDict = {}
+        for field in fieldMap:
+            if "skip_if_below" in fieldMap[field]:
+                try:
+                    if int(record[field]) < fieldMap[field]["skip_if_below"]:
+                        # Don't apply rule to records in this case (e.g. program activity before 2016)
+                        return True
+                except (TypeError, ValueError):
+                    # Could not cast as an int, this is a failure for this record
+                    return False
+            if "pad_to_length" in fieldMap[field]:
+                # Pad with leading zeros if needed
+                try:
+                    fieldValue = cls.padToLength(record[field],fieldMap[field]["pad_to_length"])
+                except ValueError as e:
+                    # If we cannot pad this value, it is not matchable (usually too long), so the rule has failed
+                    return False
+            else:
+                fieldValue = record[field]
+            valueDict[fieldMap[field]["target_field"]] = fieldValue
+
+        # Parse out model object
+        model = getattr(domainModels,str(rule.rule_text_1))
+
+        # Filter query by each field
+        query = interfaces.validationDb.session.query(model)
+        for field in valueDict:
+            query = query.filter(getattr(model,field) == valueDict[field])
+
+        # NoResultFound is return False, other exceptions should be reraised
+        try:
+            # Check that value exists in table, should be unique
+            interfaces.validationDb.runUniqueQuery(query,"Data not found in table", "Conflicting entries found for this data")
+            # If unique result found, rule passed
+            return True
+        except ResponseException as e:
+            # If exception is no result found, rule failed
+            if type(e.wrappedException) == type(NoResultFound()):
+                return False
+            else:
+                # This is an unexpected exception, so re-raise it
+                raise
+
+    @staticmethod
+    def padToLength(data,padLength):
+        """ Pad data with leading zeros
+
+        Args:
+            data: string to be padded
+            padLength: length of string after padding
+
+        Returns:
+            padded string of length padLength
+        """
+        if data is None:
+            # Convert None to empty string so it can be padded with zeros
+            data = ""
+        data = data.strip()
+
+        if len(data) < padLength:
```

Comment:
Agreed, this change was moving an existing function from before we started doing zfill for this, I've changed it to use that.